### PR TITLE
introduce CHANNELLEN and convert some strncpy + \0 into strlcpy()

### PR DESCRIPTION
**Pull request by michaelortmann**
Found by: michaelortmann
Patch by: michaelortmann
Fixes: enhancement only patch

One-line summary:
introduce CHANNELLEN and convert some strncpy + \0 into strlcpy()

Additional description (if needed):
we had some irc discussion about #3 and #38, this patch here should be fine as is.

Test cases demonstrating functionality (if applicable):

test case 1 (actually here are 2 places checked at once):
i startet eggdrop, invited it to a +i channel and it worked fine.
i also put 2 debug printf()s into source for test only and got:
DEBUG: sizeof last_invchan = 81
and
DEBUG: sizeof chan->name = 81
(the first DEBUG is for the invite, the second for the join)

test case 2:
.+chan #test
worked fine
i also put a debug printf() into source for test only and got:
DEBUG: sizeof chan->dname = 81

test case 3:
.+chan #test2
.tcl addchanrec -HQ #test2
Tcl: 1
i also put a debug printf() into source for test only and got:
DEBUG: sizeof ch->channel = 81

test case 4:
.dcc list
[02:46:06] #-HQ# dccstat
[...]
1   0.0.0.0                                       0 -HQ       chat  flags: cptEp/0
[..]
.tcl console 1 #test mkcoblxs
Tcl: {#test} mkcoblxs
i did .dcc list to find out idx = 1 parameter for the following .tcl console
i also put a debug printf() into source for test only and got:
DEBUG: sizeof dcc[i].u.chat->con_chan = 81

test case 5:
.stick
i also put a debug printf() into source for test only and got:
DEBUG: sizeof chname = 81

### CODE REVIEW

**Comment by Cizzle** [comment]
LGTM, though there's still a lot more places with the hardcoded 81.

**Comment by michaelortmann** [comment]
i did some more places.
enough for now.
because i fear, if i did even more, this pull request would become too bloated to be merged.

**Comment by michaelortmann** [comment]
i did some more places.
enough for now.
because i fear, if i did even more, this pull request would become too bloated to be merged.

**Comment by Cizzle** [comment]
LGTM

**Comment by vanosg** [comment]
build

File: src/mod/channels.mod/tclchan.c [hunk 1]
```diff
@@ -2100,8 +2100,7 @@ static int tcl_channel_add(Tcl_Interp *irp, char *newname, char *options)
      * any code later on. chan->name gets updated with the channel name as
      * the server knows it, when we join the channel. <cybah>
      */
-    strncpy(chan->dname, newname, 81);
-    chan->dname[80] = 0;
+    strncpyz(chan->dname, newname, sizeof chan->dname);
```
Comment (Cizzle) [inline src/mod/channels.mod/tclchan.c]: `strlcpy`

File: src/mod/channels.mod/userchan.c [hunk 2]
```diff
@@ -43,8 +43,7 @@ static struct chanuserrec *add_chanrec(struct userrec *u, char *chname)
     ch->flags = 0;
     ch->flags_udef = 0;
     ch->laston = 0;
-    strncpy(ch->channel, chname, 81);
-    ch->channel[80] = 0;
+    strncpyz(ch->channel, chname, sizeof ch->channel);
```
Comment (Cizzle) [inline src/mod/channels.mod/userchan.c]: `strlcpy`

File: src/mod/irc.mod/chan.c [hunk 3]
```diff
@@ -1510,8 +1510,7 @@ static int gotinvite(char *from, char *msg)
     if (now - last_invtime < 30)
       return 0; /* Two invites to the same channel in 30 seconds? */
   putlog(LOG_MISC, "*", "%s!%s invited me to %s", nick, from, msg);
-  strncpy(last_invchan, msg, 299);
-  last_invchan[299] = 0;
+  strncpyz(last_invchan, msg, sizeof last_invchan);
```
Comment (Cizzle) [inline src/mod/irc.mod/chan.c]: `strlcpy`

File: src/mod/irc.mod/chan.c [hunk 4]
```diff
@@ -1818,8 +1817,7 @@ static int gotjoin(char *from, char *chname)
           /* It was me joining! Need to update the channel record with the
            * unique name for the channel (as the server see's it). <cybah>
            */
-          strncpy(chan->name, chname, 81);
-          chan->name[80] = 0;
+          strncpyz(chan->name, chname, sizeof chan->name);
```
Comment (Cizzle) [inline src/mod/irc.mod/chan.c]: `strlcpy`

**Review by Cizzle — CHANGES_REQUESTED** [review]

File: src/mod/channels.mod/tclchan.c (hunk 1 above)
Comment (michaelortmann) [reply src/mod/channels.mod/tclchan.c]: resolved

File: src/mod/channels.mod/userchan.c (hunk 2 above)
Comment (michaelortmann) [reply src/mod/channels.mod/userchan.c]: resolved

File: src/mod/irc.mod/chan.c (hunk 3 above)
Comment (michaelortmann) [reply src/mod/irc.mod/chan.c]: resolved

File: src/mod/irc.mod/chan.c (hunk 4 above)
Comment (michaelortmann) [reply src/mod/irc.mod/chan.c]: resolved

**Comment by michaelortmann** [comment]
@Cizzle: requested changes were done.

**Review by Cizzle — APPROVED** [review]

**Comment by vanosg** [comment]
build

**Pull request merged**